### PR TITLE
Add compute instance beta tests for c3d machine type

### PR DESCRIPTION
**Pull request by hajirazaman**
C3D SEV VMs are in available in Public Preview. Add compute instance tests for c3d machine type.

Fixes b/354029489

### CODE REVIEW

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 1]
```diff
@@ -1891,6 +1891,10 @@ func TestAccComputeInstanceConfidentialInstanceConfigMain(t *testing.T) {
 
 	var instance compute.Instance
 	var instance2 compute.Instance
+	<% unless version == "ga" %>
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:1894]:
```suggestion
	<% unless version == "ga" -%>
```

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 2]
```diff
@@ -7704,9 +7714,62 @@ resource "google_compute_instance" "foobar2" {
     on_host_maintenance = "TERMINATE"
   }
 
+}
+
+<% unless version == "ga" %>
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7719]:
```suggestion
<% unless version == "ga" -%>
```

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 3]
```diff
@@ -1905,14 +1909,20 @@ func TestAccComputeInstanceConfidentialInstanceConfigMain(t *testing.T) {
 					testAccCheckComputeInstanceHasConfidentialInstanceConfig(&instance, true, "SEV"),
 					testAccCheckComputeInstanceExists(t, "google_compute_instance.foobar2", &instance2),
 					testAccCheckComputeInstanceHasConfidentialInstanceConfig(&instance2, true, ""),
+					<% unless version == "ga" %>
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:1912]:
```suggestion
					<% unless version == "ga" -%>
```

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 4]
```diff
@@ -1891,6 +1891,10 @@ func TestAccComputeInstanceConfidentialInstanceConfigMain(t *testing.T) {
 
 	var instance compute.Instance
 	var instance2 compute.Instance
+	<% unless version == "ga" %>
+	var instance3 compute.Instance
+	var instance4 compute.Instance
+	<% end %>
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:1897]:
```suggestion
	<% end -%>
```

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 5]
```diff
@@ -1905,14 +1909,20 @@ func TestAccComputeInstanceConfidentialInstanceConfigMain(t *testing.T) {
 					testAccCheckComputeInstanceHasConfidentialInstanceConfig(&instance, true, "SEV"),
 					testAccCheckComputeInstanceExists(t, "google_compute_instance.foobar2", &instance2),
 					testAccCheckComputeInstanceHasConfidentialInstanceConfig(&instance2, true, ""),
+					<% unless version == "ga" %>
+					testAccCheckComputeInstanceExists(t, "google_compute_instance.foobar3", &instance3),
+					testAccCheckComputeInstanceHasConfidentialInstanceConfig(&instance3, true, "SEV"),
+					testAccCheckComputeInstanceExists(t, "google_compute_instance.foobar4", &instance4),
+					testAccCheckComputeInstanceHasConfidentialInstanceConfig(&instance4, true, ""),
+					<% end %>
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:1917]:
```suggestion
					<% end -%>
```

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 6]
```diff
@@ -7704,9 +7714,62 @@ resource "google_compute_instance" "foobar2" {
...
+  }
+
+  scheduling {
+    on_host_maintenance = "TERMINATE"
+  }
+
 }
 `, instance, confidentialInstanceType, instance)
 }
+<% end %>
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7772]:
```suggestion
<% end -%>
```

Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb]: Done.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 5 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb]: Done.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 4 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb]: Done.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 3 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb]: Done.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 2 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb]: Done.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 1 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb]: Done.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 7]
```diff
@@ -7705,8 +7715,61 @@ resource "google_compute_instance" "foobar2" {
   }
 
 }
-`, instance, confidentialInstanceType, instance)
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7708]: Why was this line deleted?

Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7708]: That line was replaced with
`, instance, confidentialInstanceType, instance, instance, confidentialInstanceType, instance) on line 7770.

Comment (trodge) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7708]: This breaks the GA build because in GA this function no longer terminates.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb [hunk 8]
```diff
@@ -7705,7 +7715,60 @@ resource "google_compute_instance" "foobar2" {
...
+    enable_confidential_compute       = true
+  }
+
+  scheduling {
+    on_host_maintenance = "TERMINATE"
+  }
+
+}
+<% end -%>
+`, instance, confidentialInstanceType, instance, instance, confidentialInstanceType, instance)
```
Comment (trodge) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7771]: Won't this have too many arguments in the GA case? I think you need an `else` clause with the old arguments.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 7 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7708]: My bad, the version guard <% end -%> was placed incorrectly, should be fixed now.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 8 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7770]: Done, thanks.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb (hunk 7 above)
Comment (hajirazaman) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_test.go.erb:7708]: Fixed with `else` cause for old arguments.

**Review by trodge — APPROVED** [review]
